server/src/instant/storage/s3.clj

```diff
@@ -90,11 +92,19 @@
 ;; Instant <> S3 integration
 ;; ----------------------
 
-(defn upload-file-to-s3 [{:keys [app-id location-id] :as ctx} file]
+(defn upload-file-to-s3 [{:keys [app-id location-id content-type] :as ctx} file]
   (when (not (instance? java.io.InputStream file))
     (throw (Exception. "Unsupported file format")))
-  (let [ctx* (assoc ctx :object-key (->object-key app-id location-id))]
-    (s3-util/upload-stream-to-s3 (s3-async-client) bucket-name ctx* file)))
+  (let [tika-stream (TikaInputStream/get file)
+        detected-mime-type (when-not content-type
+                             (.detect (Tika.) tika-stream))
```

Comment:
I'd put this in a def, e.g.

`(def detector (Tika.))`

And then

`(.detect detector tika-tika-stream)`